src/app/components/shared/map/weightedSuperClusterAlgorithm.ts

```diff
@@ -0,0 +1,42 @@
+import { Cluster, Marker, SuperClusterAlgorithm } from "@googlemaps/markerclusterer";
+import type { ClusterFeature } from "supercluster";
+
+type MarkerWeight = number;
+
+export type WeightedClusterMarker = Marker & {
+  clusterWeight?: MarkerWeight;
+};
+
+function getMarkerWeight(marker: Marker): MarkerWeight {
+  const weightedMarker = marker as WeightedClusterMarker;
+  return weightedMarker.clusterWeight ?? 1;
+}
+
+class WeightedCluster extends Cluster {
+  public override get count(): number {
+    return this.markers.reduce((total, marker) => {
+      return total + getMarkerWeight(marker);
+    }, 0);
+  }
+}
+
+export class WeightedSuperClusterAlgorithm extends SuperClusterAlgorithm {
```

Comment:
could do with a couple of doc strings.

// gets a marker's custom clusterWeight or defaults to one

// creates a cluster that sums all the weights of the markers within it

// ....
(i don't know what the `WeightedSuperClusterAlgorithm` class does)